pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -56,6 +58,21 @@ func (a *AgentPXEFiles) Generate(dependencies asset.Parents) error {
 
 	a.imageReader = custom
 	a.cpuArch = agentArtifacts.CPUArch
+	// get the initrd.addrsize depending on the CPU arch
+	if a.cpuArch == arch.RpmArch(types.ArchitectureS390X) {
+		agentInitrdAddrFilePath := filepath.Join(a.tmpPath, "images", "initrd.addrsize")
```

Comment:
Please add integration tests to test the new files. Sample integration tests to follow are here https://github.com/openshift/installer/tree/master/cmd/openshift-install/testdata/agent/pxe/configurations

Reply:
@pawanpinjarkar: I added a integration test file (sno_s390x.txt)

@andfasano: As discussed I changed the code accordingly.

Reply:
Nice, thanks